src/main/java/org/subethamail/smtp/server/SourceTrackingSessionHandler.java

```diff
@@ -0,0 +1,96 @@
+package org.subethamail.smtp.server;
+
+import java.net.InetAddress;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * A {@link SessionHandler} to track and limit connection counts by remote addresses.
+ *
+ * @author Diego Salvi
+ */
+public class SourceTrackingSessionHandler implements SessionHandler {
```

Comment:
`SourceTrackingSessionHandler` doesn't describe its function very precisely. I'd like to capture the notion of limiting the number of concurrent sessions by source so perhaps `ConcurrentSessionsBySourceLimiter` (forget about the suffix `SessionHandler`)?

Reply:
+1 for ConcurrentSessionsBySourceLimiter... it's more clear

Reply:
In fact we could ease discovery by making the class internal and instantiate it via `SessionHandler.limitConcurrentConnectionsBySource(int maxConcurrentConnections)`.